Source/WebCore/html/canvas/WebGLUtilities.h

```diff
@@ -231,6 +231,9 @@ class ScopedWebGLRestoreTexture {
 
     ~ScopedWebGLRestoreTexture()
     {
+        if (m_context->m_textureUnits.size() <= m_context->m_activeTextureUnit)
+            return;
```

Comment:
Is this to guard against `m_context` becoming invalid? Maybe a check of `m_context` would be more appropriate?

Reply:
Yes, specifically `m_context->m_context` i.e., `m_context->graphicsContextGL()`

Reply:
That is the only way I can see how `m_activeTextureUnit` would exceed `m_textureUnits.size()` via inspection, unless I am failing to see some code path